unfurl/server.py

```diff
@@ -0,0 +1,100 @@
+import click
+from flask import Flask, request, jsonify
+from flask_caching import Cache
+import uvicorn
+
+from unfurl.localenv import LocalEnv
+
+
+flaskConfig = {
+    # Use in-memory caching, see https://flask-caching.readthedocs.io/en/latest/#built-in-cache-backends for more options
+    "CACHE_TYPE": "simple", 
+}
+unfurlConfig = {}
+
+app = Flask(__name__)
+app.config.from_mapping(flaskConfig)
+cache = Cache(app)
+
+@app.before_request
+def hook():
+    """
+    Run before every request. If the secret is specified, check all requests for the secret.
+    Secret can be in the secret query parameter (localhost:8080/health?secret=<secret>) or as an
+    Authorization bearer token (Authorization=Bearer <secret>).
+    """
+    secret = unfurlConfig["secret"]
+    if secret is None: # No secret specified, no authentication required
+        return
+
+    qs_secret = request.args.get("secret") # Get secret from query string
+    header_secret = request.headers.get('Authorization') # Get secret from Authorization header
+    if header_secret is not None:
+        try:
+            # Remove "Bearer " from header
+            header_secret = header_secret.split(' ')[1]
+        except IndexError: # Quick sanity check to make sure the header is formatted correctly
+            return jsonify({
+                "code": "BAD_REQUEST",
+                "message": "The Authorization header must be in the format 'Bearer <secret>'"
+            }), 400 # BAD_REQUEST
+
+    if secret not in [qs_secret, header_secret]: # No valid secret found in headers or qs
+        return jsonify({
+            "code": "UNAUTHORIZED",
+            "message": "Please pass the secret as a query parameter or as an Authorization bearer token"
+        }), 401 # Unauthorized
+
+
+@app.route("/health")
+def health():
+    return "OK"
+
+
+@app.route("/export")
+@cache.cached(query_string=True) # Ensure that the request cached includes the query string (the response differs between different formats)
+def export():
+    format = request.args.get("format", "deployment")
+    if format not in ["blueprint", "environments", "deployment"]:
+        return jsonify({
+            "code": "BAD_REQUEST",
+            "message": "Query parameter 'format' must be one of 'blueprint', 'environments' or 'deployment'"
+        }), 400 # BAD_REQUEST
+
+    # TODO: Check cache
```

Comment:
do we need this if there's already the request level cache?

Reply:
No, must have forgotten it.